Common/Orders/Fees/InteractiveBrokersFeeHelper.cs

```diff
@@ -0,0 +1,584 @@
+        {
+            // The futures options fee model is exactly the same as futures' fees on IB.
+            if (market == Market.Globex || market == Market.NYMEX
+                || market == Market.CBOT || market == Market.ICE
+                || market == Market.CFE || market == Market.COMEX
+                || market == Market.CME || market == Market.NYSELIFFE)
+            {
+                // just in case...
+                market = Market.USA;
+            }
+
+            if (!feeRef.TryGetValue(market, out var feeRatePerContractFunc))
+            {
+                throw new KeyNotFoundException(Messages.InteractiveBrokersFeeModel.UnexpectedFutureMarket(market));
+            }
+
+            var feeRatePerContract = feeRatePerContractFunc(security);
+            fee = quantity * feeRatePerContract.Amount;
+            currency = feeRatePerContract.Currency;
+        }
+
+        /// <summary>
+        /// Calculate the transaction fee of an Equity order
+        /// </summary>
+        /// <returns>Commission part of the transaction cost</returns>
+        internal static void CalculateEquityFee(decimal quantity, decimal tradeValue, string market,
+            decimal usFeeRate, decimal usMinimumFee, out decimal fee, out string currency)
+        {
+            EquityFee equityFee;
+            switch (market)
+            {
+                case Market.USA:
+                    equityFee = new EquityFee(Currencies.USD, feePerShare: usFeeRate, minimumFee: usMinimumFee, maximumFeeRate: 0.01m);
+                    break;
+                case Market.India:
+                    equityFee = new EquityFee(Currencies.INR, feePerShare: 0.01m, minimumFee: 6, maximumFeeRate: 20);
+                    break;
+                default:
+                    throw new KeyNotFoundException(Messages.InteractiveBrokersFeeModel.UnexpectedEquityMarket(market));
+            }
+
+            //Per share fees
+            var tradeFee = equityFee.FeePerShare * quantity;
+
+            //Maximum Per Order: equityFee.MaximumFeeRate
+            //Minimum per order. $equityFee.MinimumFee
+            var maximumPerOrder = equityFee.MaximumFeeRate * tradeValue;
+            if (tradeFee < equityFee.MinimumFee)
+            {
+                tradeFee = equityFee.MinimumFee;
+            }
+            else if (tradeFee > maximumPerOrder)
+            {
+                tradeFee = maximumPerOrder;
+            }
+
+            currency = equityFee.Currency;
+            //Always return a positive fee.
+            fee = Math.Abs(tradeFee);
+        }
+
+        /// <summary>
+        /// Calculate the transaction fee of a Cfd order
+        /// </summary>
+        internal static void CalculateCfdFee(Security security, Order order, out decimal fee, out string currency)
+        {
+            var value = Math.Abs(order.GetValue(security));
+            fee = 0.00002m * value; // 0.002%
+            currency = security.QuoteCurrency.Symbol;
```

Comment:
Whops careful! `GetValue` returns in account currency, which might not be related to quote currency. Same comment for the other calls to `GetValue` for forex & crypto